src/main/java/uk/gov/cshr/civilservant/service/OrganisationalUnitService.java

```diff
@@ -179,4 +187,25 @@ public OrganisationalUnit deleteAgencyToken(OrganisationalUnit organisationalUni
     public Optional<OrganisationalUnit> get(Long id) {
         return repository.findById(id);
     }
+
+    public List<String> getOrganisationalUnitCodesForIds(List<String> organisationIds) {
+        return repository.findAllCodesForIds(organisationIds);
+    }
+
+    public List<Long> getOrganisationIdWithChildrenIds(List<String> listOrganisationCodes) {
+        List<Long> listOrganisationIds = new ArrayList<>();
+        for (String code : listOrganisationCodes) {
+            getOrganisationWithChildren(code).forEach(x -> listOrganisationIds.add(x.getId()));
+        }
+        return listOrganisationIds;
```

Comment:
This method could be implemented using Java 8 Stream APIs.

Reply:
There are always opinions about using iteration or stream. stream has better performance when the data needs to be processed in pipeline like reading lines of file, reading pipeline of records from table however once all records are in variable then its better to process with iterator pattern.
Happy to discuss further.

Reply:
As discussed further over the phone, this code should be written using stream.